### PR TITLE
Replace 'serve' package w/ express-static to restore compression

### LINKED ISSUE #7119: Replace "serve" package w/ express for "gatsby serve"
`serve` recently took away many of the nice things about the package e.g. included gzip support.

We want that.

So let's just switch to express w/ a compression middleware.

### PR DESCRIPTION
**Pull request by KyleAMathews**
fixes #7119

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview for *using-postcss-sass* failed.

Built with commit 9cb60d4ebd9fcbfa94a38e6230904a20670043cf

https://app.netlify.com/sites/using-postcss-sass/deploys/5b6ab2f9c6aed62b02841c1f

**Comment by KyleAMathews** [comment]
Sorry about the Prettier noise

File: packages/gatsby/src/commands/serve.js [hunk 1]
```diff
@@ -3,18 +3,18 @@ const handler = require(`serve-handler`)
 const openurl = require(`opn`)
 const http = require(`http`)
 const signalExit = require(`signal-exit`)
+const compression = require(`compression`)
```
Comment (KyleAMathews) [inline packages/gatsby/src/commands/serve.js]: This file has the actual changes

**Comment by m-allanson** [comment]
Oh - remove `serve` from package.json?

File: packages/gatsby/src/commands/serve.js [hunk 2]
```diff
@@ -3,18 +3,18 @@ const handler = require(`serve-handler`)
 const openurl = require(`opn`)
```
Comment (m-allanson) [inline packages/gatsby/src/commands/serve.js]: and remove Serve's `serve-handler` here?

File: packages/gatsby/src/commands/serve.js [hunk 3]
```diff
@@ -3,18 +3,18 @@ const handler = require(`serve-handler`)
...
 const http = require(`http`)
```
Comment (tryzniak) [inline packages/gatsby/src/commands/serve.js]: This can be removed I guess

**Comment by m-allanson** [comment]
Could add pathPrefix handling in here to fix #5271? 

https://github.com/gatsbyjs/gatsby/issues/5271#issuecomment-386866079

**Comment by azdanov** [comment]
I was experimenting on my own with `http-server`.

Is this viable alternative? It supports quite a few [options](https://github.com/indexzero/http-server/blob/94329688b5bf6c058a9f2c24739bd628b7345dce/bin/http-server#L20) out of the box.

<details>

<summary>serve.js</summary>

```js
/* @flow weak */
const openurl = require(`opn`)
const signalExit = require(`signal-exit`)
const { createServer } = require(`http-server/lib/http-server`)

module.exports = program => {
  let { port, open } = program
  port = typeof port === `string` ? parseInt(port, 10) : port

  let server = createServer({ root: `public` })

  server.listen(port, () => {
    let openUrlString = `http://localhost:` + port
    console.log(`gatsby serve running at:`, openUrlString)
    if (open) {
      let openUrlString = `http://localhost:` + port
      console.log(`Opening browser...`)
      openurl(openUrlString)
    }
  })

  signalExit((code, signal) => {
    server.close()
  })
}
```

</details>

**Comment by KyleAMathews** [comment]
It seems like it'd work great too! I'm not overly worried about which one we use as they're essentially all the same thing running the same node http server underneath.

**Comment by KyleAMathews** [comment]
Deploy preview for *gatsbyjs* failed.

Built with commit 9cb60d4ebd9fcbfa94a38e6230904a20670043cf

https://app.netlify.com/sites/gatsbyjs/deploys/5b6ab2f8c6aed62b02841c13

**Pull request merged**